### ISSUE TITLE
[Question] Client key compiled in binary?

### ISSUE THREAD

**Issue by pwFoo** [NHAS/reverse_ssh#107]
Hi,
I played arround with server / client binaries from releases. Downloaded binaries not working for me. Client can't connet to the server.
It works fine if I compiled my own server and client binaries.

1. So I need to compile my own binaries to use personalized / fresh generated private & public keys?
2. Should I use the same client binary for different devices or compile new client binaries by `link --name <IDENTIFIER>`?

I play with small linux devices without sshd service running and a dynamic public ip address or maybe behind a firewall / nat. 
So reverse_ssh maybe is a nice way to connect devices to the server / jump host to access a secure way without public open ports.

**Comment by NHAS** [comment]
You know. You're the first person to point out I don't publish the public keys for the client binaries which are kind of important for adding them to the server authorised key...

Bit of an over site on my behalf. You can run the server with `--insecure` which would enable any client to connect. But if recommend building it yourself either via the link command or the make file.


For future releases I'll make sure to include the public key for the clients so you can use it properly.

**Comment by pwFoo** [comment]
Hi,
thanks for the quick reply. I will explain it a little more.

I have tested similar projects before and one worked with files for private key and public key of server, client and with an authorized_keys. So in the first moment I was looking where to put my generated private / public pair so that it will be read by the binaries.

I had found the solution myself to simply compile my own binaries, but maybe a note in the readme would be good, that the key pair is compiled in the client binary? And the hint to use `--insecure` to test with pre-compiled binaries?

For my needs I'll just compile custom server and client binaries to use.

Thanks a lot for creating and maintaining this great project!
